contrib/storage-xml/src/main/java/org/apache/drill/exec/store/easy/xml/XMLFormatPlugin.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.exec.ops.FragmentContext;
+import org.apache.drill.exec.proto.ExecProtos.FragmentHandle;
+import org.apache.drill.exec.proto.UserBitShared.CoreOperatorType;
+import org.apache.drill.exec.server.DrillbitContext;
+import org.apache.drill.exec.store.RecordReader;
+import org.apache.drill.exec.store.RecordWriter;
+import org.apache.drill.exec.store.dfs.DrillFileSystem;
+import org.apache.drill.exec.store.dfs.FileSystemConfig;
+import org.apache.drill.exec.store.dfs.easy.EasyFormatPlugin;
+import org.apache.drill.exec.store.dfs.easy.EasyWriter;
+import org.apache.drill.exec.store.dfs.easy.FileWork;
+import org.apache.drill.exec.store.easy.xml.XMLFormatPlugin.XMLFormatConfig;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.drill.exec.store.easy.xml.XMLRecordReader;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+
+/**
+ * Created by mpierre on 15-11-04.
+ */
+
+public class XMLFormatPlugin extends EasyFormatPlugin<XMLFormatConfig> {
+
+    private static final boolean IS_COMPRESSIBLE = false;
+    private static final String DEFAULT_NAME = "xml";
+    private Boolean keepPrefix = true;
+    private XMLFormatConfig xmlConfig;
+
+    public XMLFormatPlugin(String name, DrillbitContext context, Configuration fsConf, StoragePluginConfig storageConfig) {
+        this(name, context, fsConf, storageConfig, new XMLFormatConfig());
+    }
+
+    public XMLFormatPlugin(String name, DrillbitContext context, Configuration fsConf, StoragePluginConfig config, XMLFormatConfig formatPluginConfig) {
+        super(name, context, fsConf, config, formatPluginConfig, true, false, false, IS_COMPRESSIBLE, formatPluginConfig.getExtensions(), DEFAULT_NAME);
+        xmlConfig = formatPluginConfig;
+    }
+
+    @Override
+    public RecordReader getRecordReader(FragmentContext context, DrillFileSystem dfs, FileWork fileWork,
+                                        List<SchemaPath> columns, String userName) throws ExecutionSetupException {
+        return new XMLRecordReader(context, fileWork.getPath(), dfs, columns, xmlConfig);
+    }
+
+
+    @Override
+    public int getReaderOperatorType() {
+        return CoreOperatorType.JSON_SUB_SCAN_VALUE;
+    }
+
+    @Override
+    public int getWriterOperatorType() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean supportsPushDown() {
+        return true;
+    }
+
+    @Override
+    public RecordWriter getRecordWriter(FragmentContext context, EasyWriter writer) throws IOException {
+        return null;
+    }
+
+    @JsonTypeName("xml")
+    public static class XMLFormatConfig implements FormatPluginConfig {
+
+        public List<String> extensions;
```

Comment:
This is a bit of a lingering cleanup task, but the current formats don't share code for handling extensions. If you have some time if you would like to try to move this to a superclass (maybe put an abstract class between the interface and this one) and pull out the common code in the other format plugins that would be better than duplicating our bad pattern here.